src/utils.jl

```diff
@@ -176,94 +182,93 @@ tryrationalize(::Type{R}, x) where {R} = isinteger(x) ? convert(R, round(Int, x)
 
 Base.showerror(io::IO, e::DimensionError) = print(io, "DimensionError: ", e.q1, " and ", e.q2, " have incompatible dimensions")
 
-# TODO: these are redundant with the constructors
-Base.convert(::Type{Q}, q::AbstractQuantity) where {Q<:AbstractQuantity} = q
-Base.convert(::Type{Q}, q::AbstractQuantity) where {T,Q<:AbstractQuantity{T}} = new_quantity(Q, convert(T, ustrip(q)), dimension(q))
-Base.convert(::Type{Q}, q::AbstractQuantity) where {T,D,Q<:AbstractQuantity{T,D}} = new_quantity(Q, convert(T, ustrip(q)), convert(D, dimension(q)))
+Base.convert(::Type{Q}, q::AbstractUnionQuantity) where {Q<:AbstractUnionQuantity} = q
+Base.convert(::Type{Q}, q::AbstractUnionQuantity) where {T,Q<:AbstractUnionQuantity{T}} = new_quantity(Q, convert(T, ustrip(q)), dimension(q))
+Base.convert(::Type{Q}, q::AbstractUnionQuantity) where {T,D,Q<:AbstractUnionQuantity{T,D}} = new_quantity(Q, convert(T, ustrip(q)), convert(D, dimension(q)))
 
 Base.convert(::Type{D}, d::AbstractDimensions) where {D<:AbstractDimensions} = d
 Base.convert(::Type{D}, d::AbstractDimensions) where {R,D<:AbstractDimensions{R}} = D(d)
 
 Base.copy(d::D) where {D<:AbstractDimensions} = map_dimensions(copy, d)
-Base.copy(q::Q) where {Q<:AbstractQuantity} = new_quantity(Q, copy(ustrip(q)), copy(dimension(q)))
+Base.copy(q::Q) where {Q<:AbstractUnionQuantity} = new_quantity(Q, copy(ustrip(q)), copy(dimension(q)))
 
 """
-    ustrip(q::AbstractQuantity)
+    ustrip(q::AbstractUnionQuantity)
```

Comment:
A bit of a premature comment since there are no docs yet, but if `AbstractUnionQuantity` is used in the API description (as it probably should), then it should be clearly documented to the user (and maybe exported?)

Reply:
Good point! Maybe in the docs it can be split into separate cases for `AbstractQuantity` and `AbstractGenericQuantity` so that users don't get confused.